src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -76,6 +76,12 @@ public Renderer(IServiceProvider serviceProvider, ILoggerFactory loggerFactory)
         /// </summary>
         public abstract Dispatcher Dispatcher { get; }
 
+        /// <summary>
+        /// Gets or sets the <see cref="Components.ElementReferenceContext"/> associated with this <see cref="Renderer"/>,
+        /// if it exists.
+        /// </summary>
+        public ElementReferenceContext? ElementReferenceContext { get; protected set; }
```

Comment:
I'm ok with this because the renderer is a fairly low level type, but you *might* be able to make this `protected internal` to avoid exposing it. It's not really useful to consumers.